pkg/model/utxo/receipt.go

```diff
@@ -0,0 +1,188 @@
+package utxo
+
+import (
+	"encoding/binary"
+
+	"github.com/gohornet/hornet/pkg/model/hornet"
+	"github.com/gohornet/hornet/pkg/model/milestone"
+	"github.com/iotaledger/hive.go/kvstore"
+	"github.com/iotaledger/hive.go/marshalutil"
+	iotago "github.com/iotaledger/iota.go/v2"
+)
+
+// ReceiptTuple contains a receipt and the index of the milestone
+// which contained the receipt.
+type ReceiptTuple struct {
+	// The actual receipt.
+	Receipt *iotago.Receipt
+	// The index of the milestone which included the receipt.
+	MilestoneIndex milestone.Index
+}
+
+func (rt *ReceiptTuple) kvStorableKey() (key []byte) {
+	return marshalutil.New(9).
+		WriteByte(UTXOStoreKeyPrefixReceipts).
+		WriteUint32(rt.Receipt.MigratedAt).
+		WriteUint32(uint32(rt.MilestoneIndex)).
+		Bytes()
+}
+
+func (rt *ReceiptTuple) kvStorableValue() (value []byte) {
+	receiptBytes, err := rt.Receipt.Serialize(iotago.DeSeriModeNoValidation)
+	if err != nil {
+		panic(err)
+	}
+	return receiptBytes
+}
+
+func (rt *ReceiptTuple) kvStorableLoad(_ *Manager, key []byte, value []byte) error {
+	keyExt := marshalutil.New(key)
+
+	// skip prefix and migrated at index
+	if _, err := keyExt.ReadByte(); err != nil {
+		return err
+	}
+
+	if _, err := keyExt.ReadUint32(); err != nil {
+		return err
+	}
+
+	// read out index of the milestone which contained this receipt
+	msIndex, err := keyExt.ReadUint32()
+	if err != nil {
+		return err
+	}
+
+	r := &iotago.Receipt{}
+	if _, err := r.Deserialize(value, iotago.DeSeriModeNoValidation); err != nil {
+		return err
+	}
+
+	rt.Receipt = r
+	rt.MilestoneIndex = milestone.Index(msIndex)
+
+	return nil
+}
+
+// StoreReceipt stores the given receipt tuple.
```

Comment:
StoreReceipt and DeleteReceipt can be removed. They are unused and are not inside the batched mutations, so should be avoided.